app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,21 @@
+class PiecesController < ApplicationController
+  
+  def update
+    if selected_piece.valid_move?(params[:piece][:x_position].to_i,params[:piece][:y_position].to_i)
+      selected_piece.update_attributes(piece_params)
+      render json: { response: "Valid Move"}
```

Comment:
With JSON, the text response is irrelevant unless it's data. Success/Fail should be based solely on the response type returned.